rpcs3/rpcs3qt/hex_validator.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <QValidator>
+#include <QRegularExpression>
+
+class HexValidator : public QValidator
+{
+public:
+    explicit HexValidator(int maxNibbles, QObject *parent = nullptr)
+      : QValidator(parent)
+      , m_maxNibbles(maxNibbles)
+    {}
+
+    State validate(QString &input, int &pos) const override
+    {
+        Q_UNUSED(pos);
+
+        QString stripped = input;
+        stripped.remove(' ');
+
+        if (stripped.startsWith("0x", Qt::CaseInsensitive)) 
+            stripped = stripped.mid(2);
+
+        if (stripped.isEmpty())
+            return QValidator::Intermediate;
+
+        static const QRegularExpression hexRe("^[0-9A-Fa-f]*$");
+        if (!hexRe.match(stripped).hasMatch())
+            return QValidator::Invalid;
+
+        if (stripped.length() > m_maxNibbles)
```

Comment:
It should allow as many as 16, minus the rest of digits, leading zeroes irregardless of the context, even if it would be non-fitting for the specific input textbox.

Reply:
Are you suggesting to limit max nibbles to 16?

Reply:
No, it should only 64-bit-like values from conversions which can be expressed in lower amount of bits. For example:

0x0000 0000 0000 0001

is technically 64 bits, 16 digits.
but it can be expressed in 8 bits:

0x01

without losing meaning.

It should allow it specifically for leading zeroes.
So even in an 8-bit textbox:

0x0000 0000 0000 0001 should be allowed.
while 0xffff should not.

Reply:
Alright, I think it should be good to go now.